torch_xla/experimental/custom_kernel.py

```diff
@@ -56,38 +56,43 @@ def jax_import_guard():
   torch_xla._XLAC._init_computation_client()
 
 
-def to_jax_shape_dtype_struct(tensor: torch.Tensor) -> "jax.ShapeDtypeStruct":
+def convert_torch_dtype_to_jax(dtype: torch.dtype) -> "jnp.dtype":
   # Import JAX within the function such that we don't need to call the jax_import_guard()
   # in the global scope which could cause problems for xmp.spawn.
   jax_import_guard()
-  import jax
   import jax.numpy as jnp
 
-  def convert_torch_dtype_to_jax(dtype: torch.dtype) -> jnp.dtype:
-    if dtype == torch.float32:
-      if _XLA_USE_BF16:
-        return jnp.bfloat16
-      return jnp.float32
-    elif dtype == torch.float64:
-      if _XLA_USE_BF16:
-        return jnp.bfloat16
-      return jnp.float64
-    elif dtype == torch.float16:
-      return jnp.float16
-    elif dtype == torch.bfloat16:
+  if dtype == torch.float32:
+    if _XLA_USE_BF16:
```

Comment:
.. we should really try to just cast everything to bf16 manually after we are done with current deadline...

Reply:
This is just for backward compatibility....